lib/oban/migrations.ex

```diff
@@ -148,15 +172,24 @@ defmodule Oban.Migrations do
     end
   end
 
-  defp change(prefix, range, direction) do
+  defp change(range, direction, opts) do
     for index <- range do
       pad_idx = String.pad_leading(to_string(index), 2, "0")
 
+      opts =
+        if direction == :up && index == 1 do
+          opts
+        else
+          Keyword.take(opts, [:prefix])
+        end
+
```

Comment:
This is just about as awkward as the previous multi-arity version. The migration format isn't a public API, and we're free to change what gets passed through. What about using `Map.new(opts)` so that the various functions can reliably match on the `prefix`?

Reply:
Agreed. I thought you were keen on using Keyword for `opts` :)
But no need to use `Map.new`. I suggest we have `up` and `down` directly pass a map of options to `change`. I pushed the new changes, please take a look.

Reply:
For public APIs I definitely prefer keywords for opts. For internal abstractions I'm ambivalent, and maps make much more sense for destructuring.